### PR TITLE
vSphere: Update terraform provider to current upstream

### PR DESCRIPTION
**Pull request by jcpowermac**

### CODE REVIEW

**Comment by jcpowermac** [comment]
Currently testing in VMC but also want to make sure this works in IBM.
So far its looking surprisingly good

**Comment by jcpowermac** [comment]
We would no longer need to carry the openshift/terraform-vsphere-provider fork if this works.

**Comment by jcpowermac** [comment]
Test failures unrelated to PR. Cluster installed and passed terraform steps. My manual test in VMC installed as well without issue.

**Comment by jcpowermac** [comment]
Tested (passed) in IBM via CI: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_installer/5694/pull-ci-openshift-installer-master-e2e-vsphere/1501725755468419072

cc: @patrickdillon

**Comment by staebler** [comment]
@jcpowermac Can you build this on top of https://github.com/openshift/installer/pull/5672? The directory of the terraform providers is changing in that PR.

File: terraform/vsphere/go.mod [hunk 1]
```diff
@@ -2,71 +2,68 @@ module github.com/openshift/installer/terraform/vsphere
...
+	golang.org/x/tools v0.0.0-20201028111035-eafbe7b904eb // indirect
+	golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1 // indirect
+	google.golang.org/api v0.34.0 // indirect
+	google.golang.org/appengine v1.6.6 // indirect
+	google.golang.org/genproto v0.0.0-20200904004341-0bd0a958aa1d // indirect
+	google.golang.org/grpc v1.32.0 // indirect
+	google.golang.org/protobuf v1.25.0 // indirect
 )
-
-replace github.com/hashicorp/terraform-provider-vsphere => github.com/openshift/terraform-provider-vsphere v1.24.3-openshift
```
Comment (staebler) [inline terraform/vsphere/go.mod:72]: :woman_dancing:

**Comment by jcpowermac** [comment]
> > We would no longer need to carry the openshift/terraform-vsphere-provider fork if this works.
> 
> @jcpowermac I still see [hashicorp/terraform-provider-vsphere#1028](https://github.com/hashicorp/terraform-provider-vsphere/issues/1028) as an open issue. Do we not need to carry the patch for it any more?

I didn't hit issue 1028 like previously and when we experienced that problem previously it was immediate in the virtual machine create process. So I am thinking no.

**Comment by jcpowermac** [comment]
> @jcpowermac Can you build this on top of #5672? The directory of the terraform providers is changing in that PR.

Sure

**Comment by tenthirtyam** [comment]
@jcpowermac We never committed https://github.com/hashicorp/terraform-provider-vsphere/pull/1075. Let us know if that one needs to be closed and if your tests continue to show good results. I assume there has been a TF bump as well on your side?

**Comment by jcpowermac** [comment]
> @jcpowermac We never committed [hashicorp/terraform-provider-vsphere#1075](https://github.com/hashicorp/terraform-provider-vsphere/pull/1075) based on [your previous comment](https://github.com/hashicorp/terraform-provider-vsphere/pull/1075#issuecomment-631599375). Let us know if that one needs to be closed and if your tests continue to show good results. I assume there has been a TF bump as well on your side?

We have our own fork: https://github.com/openshift/terraform-provider-vsphere where we carried that commit.
I am using this PR to test and see if we still need it and so far its looking good.

Went ahead and closed [hashicorp/terraform-provider-vsphere#1075](https://github.com/hashicorp/terraform-provider-vsphere/pull/1075) as I don't think its necessary anymore.

**Pull request merged**